### ISSUE TITLE
Map Rendering

### ISSUE THREAD

**Issue by Septharoth** [ethanmoffat/EndlessClient#21]
Just AFK on a map with NPCs (and left-over item drops.. not sure if this could also be an issue) you're likely to crash.

I actually can't even login on my character when I log out on this one map on my server!  (Has 16 Biters, and random golds, small potions left on the ground from players training here)

**Comment by ethanmoffat** [comment]
a) Have you run it in the debugger at all? So you can see where exactly it's crashing?
b) What's the map ID of the map you're stuck on?

**Comment by Septharoth** [comment]
a) I have not run the debugger.  Will do that now, didn't even think about that.. silly me.

b) it's a custom map on my server, map id 503.  I loaded on the map fine, but once I stood by the NPCs and a player was killing them, I started crashing.

**Comment by ethanmoffat** [comment]
Okay, yeah, run it in the debugger and see where it crashes. I'll look into making a global exception handler that logs relevant information into the debug\log and throws you back to the main menu instead of crashing the game.

**Comment by ethanmoffat** [comment]
Try it with commits 2874f0386e9370eb0fbec1cabf7721b3f0087c7c and db612e5c45b9ae74ca27cbe6c590b80b99e37dc6. I found a bug in the packet decoding function. I had worked around it before but it has only popped up in two places so far during the year or so I've been working on this.

The first commit fixes the decoding function, the second adds a global exception handler that does some logging when it crashes (debug builds only).
